python/lsst/ts/mtdome/mock_llc/apscs.py

```diff
@@ -87,85 +139,185 @@ async def determine_status(self, current_tai: float) -> None:
         }
         self.log.debug(f"apcs_state = {self.llc_status}")
 
-    async def openShutter(self) -> None:
-        """Open the shutter."""
-        self.command_time_tai = utils.current_tai()
-        self.status = [LlcMotionState.OPEN, LlcMotionState.OPEN]
-        # Both positions are expressed in percentage.
-        self.position_actual = np.full(NUM_SHUTTERS, 100.0, dtype=float)
-        self.position_commanded = 100.0
-
-    async def closeShutter(self) -> None:
-        """Close the shutter."""
-        self.command_time_tai = utils.current_tai()
-        self.status = [LlcMotionState.CLOSED, LlcMotionState.CLOSED]
-        # Both positions are expressed in percentage.
-        self.position_actual = np.zeros(NUM_SHUTTERS, dtype=float)
-        self.position_commanded = 0.0
+    async def openShutter(self, start_tai: float) -> float:
+        """Open the shutter.
 
-    async def stopShutter(self) -> None:
-        """Stop all motion of the shutter."""
-        self.command_time_tai = utils.current_tai()
-        self.status = [LlcMotionState.STOPPED, LlcMotionState.STOPPED]
+        Parameters
+        ----------
+        start_tai: `float`
+            The TAI time, unix seconds, when the command was issued. To model
+            the real dome, this should be the current time. However, for unit
+            tests it can be convenient to use other values.
 
-    async def go_stationary(self) -> None:
-        """Stop shutter motion and engage the brakes."""
-        self.command_time_tai = utils.current_tai()
-        self.status = [LlcMotionState.STATIONARY, LlcMotionState.STATIONARY]
+        Returns
+        -------
+        `float`
+            The expected duration of the command [s].
+        """
+        self.position_commanded = OPEN_POSITION
+        duration = 0.0
+        for i in range(NUM_SHUTTERS):
+            duration = self.shutter_motion[i].set_shutter_target_position_and_velocity(
+                start_tai=start_tai,
+                end_position=self.position_commanded,
+                motion_state=LlcMotionState.MOVING,
+            )
+        return duration
```

Comment:
Yes this is possible. I was thinking about this and wanted to discuss with the manufacturer. This will also be an issue with the louvers. So I opened a ticket for it and added TODO's.